### PR TITLE
Add explosions

### PR DESCRIPTION
**Pull request by BlakeMScurr**
Closes https://github.com/TruncateGame/Truncate/issues/8

### CODE REVIEW

File: core/src/moves.rs [hunk 1]
```diff
@@ -120,6 +121,23 @@ impl Board {
...
+                                return self.clear(square).map(|detail| {
+                                    Change::Board(BoardChange {
+                                        detail,
+                                        action: BoardChangeAction::Exploded,
+                                    })
+                                });
+                            }
+                        }
+                        None
+                    }));
```
Comment (bglw) [inline core/src/moves.rs:140]: Could this instead rely on the existing `board.neighbouring_squares(position)`?

https://github.com/TruncateGame/Truncate/blob/c47641ff3efeaf47190a2ad512f4aa385bf1d1be/src/board.rs#L173-L184

**Review by bglw — CHANGES_REQUESTED** [review]

File: core/src/moves.rs (hunk 1 above)
Comment (BlakeMScurr) [reply core/src/moves.rs]: Yeah, I think it's probably clearer (though about equally verbose).

**Review by bglw — APPROVED** [review]
Need to make a note to render the new Exloded status in the client

**Comment by BlakeMScurr** [comment]
Ah good point, I'll give that a go as part of https://github.com/TruncateGame/Truncate/issues/8

**Pull request merged**